servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -307,9 +307,28 @@ HttpServerBuilder appendServiceFilter(Predicate<StreamingHttpRequest> predicate,
     HttpServerBuilder bufferAllocator(BufferAllocator allocator);
 
     /**
-     * Sets the {@link HttpExecutionStrategy} to be used by this server.
+     * Sets the base {@link HttpExecutionStrategy} to be used by this server. If not provided then the default safe
+     * base strategy appropriate for the server type (async/blocking, aggregate/streaming) will be used. The strategy
+     * used by the server may also be influenced by the service or filters if either requires additional offloading.
      *
-     * @param strategy {@link HttpExecutionStrategy} to use by this server.
+     * <p>Some examples of the effective server strategy used for various scenarios:</p>
+     *  builder:                service:                  filters:             Effective:
+     *  defaultStrategy()
+     *  (if service or filter needs additional offloading, add it)
+     *                          offloadNone()             offloadNone()        apiDefaultStrategy()
+     *                          customStrategy()          offloadNone()        apiDefaultStrategy() + customStrategy()
+     *                          offloadNone()             customStrategy()     apiDefaultStrategy() + customStrategy()
+     *  customStrategy()
+     *  (if service or filter needs additional offloading, add it)
+     *                          offloadNone()             offloadNone()        customStrategy()
+     *                          customStrategy()          offloadNone()        customStrategy() + customStrategy()
```

Comment:
This change looks unrelated to the original problem this PR is trying to solve and increases review iterations. Consider taking it out to a separate PR.
These abbreviations are still hard to understand for new users.

Reply:
Sorry, I should have moved the PR in to draft before pushing those changes. I have replaced the PR with a different more minimal one.